pkg/cmd/roachtest/test_runner.go

```diff
@@ -1736,56 +1745,90 @@ func (r *testRunner) teardownTest(
-			t.appendGithubMessage(out)
-			return nil
+				"will be shown here. See node logs in artifacts for more details.\n%s", fatalOut)
+			t.appendGithubFatalLogs(fatalOut)
+		}
+
+		t.L().Printf("Attempting to gather ip node mapping")
+		ipNodeMapOut, gatherNodeIpErr := gatherNodeIpMapping(t, c)
+		if gatherNodeIpErr != nil {
+			inspectArtifactsErr = errors.Join(inspectArtifactsErr, gatherNodeIpErr)
+		} else {
+			t.appendGithubIpToNodeMapping(ipNodeMapOut)
 		}
 	}
-	return nil
+	return inspectArtifactsErr
+}
+
+// gatherNodeIpMapping attempts to gather cluster node ip information for debug
+func gatherNodeIpMapping(t *testImpl, c *clusterImpl) (string, error) {
+	var table [][]string
+	cachedCluster, err := getCachedCluster(c.name)
+	if err != nil {
+		return "", err
+	}
+	// The regex to select out this table in issues.roachtestNodeToIpRE is not
+	// strict on the number of columns i.e. columns can be added / removed
+	// without a regex change
+	table = append(table, []string{"Node", "Private IP", "Public IP", "Machine Type"})
+	for _, vmInstance := range cachedCluster.VMs {
+		table = append(table, []string{vmInstance.Name, vmInstance.PrivateIP, vmInstance.PublicIP, vmInstance.MachineType})
+	}
+	nodeIpTable, err := roachtestutil.ToMarkdownTable(table)
+	if err != nil {
+		return "", err
+	}
+	testClusterLogger, err := c.l.ChildLogger("node-ips", logger.QuietStderr, logger.QuietStdout)
+	if err != nil {
+		t.L().Printf("unable to create logger %s: %s", "node-ips", err)
+		return "", err
+	}
+	testClusterLogger.Printf("\n%s", nodeIpTable)
+	return nodeIpTable, nil
 }
 
 // gatherFatalNodeLogs attempts to gather fatal level node logs to help with
 // triage
 func gatherFatalNodeLogs(t *testImpl, testLogger *logger.Logger) (string, error) {
 	logPattern := `^F[0-9]{6}`
 	filePattern := "logs/*unredacted/cockroach*.log"
-	// *unredacted captures patterns for single node and multi-node clusters
-	// e.g. unredacted, 1.unredacted
+	// * wildcard to capture patterns for single node and multi-node clusters
+	// e.g. single node: unredacted, multi-node: 1.unredacted, ...
 	joinedFilePath := filepath.Join(t.ArtifactsDir(), filePattern)
 	targetFiles, err := filepath.Glob(joinedFilePath)
 	if err != nil {
 		return "", err
 	} else if len(targetFiles) == 0 {
-		return "", errors.Newf("No matching log files found for log pattern: %s and file pattern: %s",
+		t.L().Printf("No matching log files found for log pattern: %s and file pattern: %s",
 			logPattern, filePattern)
+		return "", nil
 	}
 	args := append([]string{"-E", "-m", "10", "-a", logPattern}, targetFiles...)
 	command := "grep"
 	t.L().Printf("Gathering fatal level logs with command: %q %s", command, strings.Join(args, " "))
 	// Works with local and remote node clusters because we will always download
 	// the artifacts if there's a test failure (except for timeout)
-	cmd := exec.Command("grep", args...)
+	cmd := exec.Command(command, args...)
```

Comment:
nit: change seems redundant since command is never modified, makes it marginally harder to follow